pkg/staking/stakingcontract/contract.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/sctx"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	MinimumStakeAmount = big.NewInt(1)
+
+	erc20ABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: get ABI for staking contract and replace it below
+	stakingABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+
+	ErrInsufficientStakeAmount = errors.New("insufficient stake amount")
+	ErrInsufficientFunds       = errors.New("insufficient token balance")
+	ErrNotImplemented          = errors.New("not implemented")
+
+	depositStakeDescription = "Deposit Stake"
+)
+
+type Interface interface {
+	DepositStake(ctx context.Context, stakedAmount *big.Int, overlay swarm.Address) error
+	GetStake(ctx context.Context, overlay swarm.Address) (*big.Int, error)
+}
+
+type contract struct {
+	owner                  common.Address
+	stakingContractAddress common.Address
+	bzzTokenAddress        common.Address
+	transactionService     transaction.Service
+	overlayNonce           common.Hash
+}
+
+func New(
+	owner common.Address,
+	stakingContractAddress common.Address,
+	bzzTokenAddress common.Address,
+	transactionService transaction.Service,
+	nonce common.Hash,
+) Interface {
+	return &contract{
+		owner:                  owner,
+		stakingContractAddress: stakingContractAddress,
+		bzzTokenAddress:        bzzTokenAddress,
+		transactionService:     transactionService,
+		overlayNonce:           nonce,
+	}
+}
+
+func (s *contract) sendTransaction(ctx context.Context, callData []byte, desc string) (*types.Receipt, error) {
+	gasLimit := sctx.GetGasLimit(ctx)
+	if gasLimit == 0 {
+		// if gas limit is not set, use the default limit.
+		gasLimit = 3_000_000
+	}
+	request := &transaction.TxRequest{
+		To:          &s.stakingContractAddress,
+		Data:        callData,
+		GasPrice:    sctx.GetGasPrice(ctx),
+		GasLimit:    gasLimit,
+		Value:       big.NewInt(0),
+		Description: desc,
+	}
+
+	txHash, err := s.transactionService.Send(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := s.transactionService.WaitForReceipt(ctx, txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if receipt.Status == 0 {
+		return nil, transaction.ErrTransactionReverted
+	}
+
+	return receipt, nil
+}
+
+func (s *contract) sendDepositStakeTransaction(ctx context.Context, owner common.Address, stakedAmount *big.Int, nonce common.Hash) (*types.Receipt, error) {
+	callData, err := stakingABI.Pack("depositStake", owner, stakedAmount, nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := s.sendTransaction(ctx, callData, depositStakeDescription)
+	if err != nil {
+		return nil, fmt.Errorf("deposit stake: stakedAmount %d: %w", stakedAmount, err)
+	}
+
+	return receipt, nil
+}
+
+func (s *contract) sendGetStakeTransaction(ctx context.Context, overlay swarm.Address) (*big.Int, error) {
```

Comment:
So this function naming is not correct. This is not a transaction. This is similar to the helper `getBalance`, so it should be called just `getStake`.